Adding hashmap class

TatianaCarrascoOrtiz opened the pull request:
First version on HashMap class to umltxt

TatianaCarrascoOrtiz committed:
Adding hashmap class

budaimartin commented on src/hu/elte/txtuml/stdlib/util/HashMap/Hashmap.java:
```diff
@@ -0,0 +1,99 @@
+ 	 * Removes the mapping for the specified key from this map if present.
+ 	 *
+ 	 */
+ 	 public HashMap<String, String> remove(Object key);
```
Parameter `key` should be `String` since `Object` is not a txtUML class nor an external class. And why is the return value `HashMap<String, String>`?

budaimartin commented on src/hu/elte/txtuml/stdlib/util/HashMap/Hashmap.java:
```diff
@@ -0,0 +1,99 @@
+ 	 * If the map previously contained a mapping for the key, the old value is replaced. 
+ 	 *
+ 	 */
+ 	 public HashMap<String, String> put(String key, String value);
```
Why is the return value `HashMap<String, String>`?

budaimartin reviewed (changes requested):
Please delete the imports in comments and unnecessary blank lines. Also run autoformat on your code before re-uploading. See my other comments below.

TatianaCarrascoOrtiz committed:
Update EntryImplementation.java

Deleting comments

TatianaCarrascoOrtiz committed:
Update Hashmap.java

Deleting comments

TatianaCarrascoOrtiz committed:
Update Entry.java

Deleting spaces and comments

TatianaCarrascoOrtiz committed:
Update EntryImplementation.java

Deleting comments

TatianaCarrascoOrtiz committed:
Update Hashmap.java

Deleting comments

TatianaCarrascoOrtiz committed:
Update Hashmap.java

Deleting spaces

TatianaCarrascoOrtiz committed:
Update HashmapImplementation.java

Deleting spaces and comments

TatianaCarrascoOrtiz committed:
Update Hashmap.java

The return value is  HashMap<String, String>, because with the teacher decided that only we will work with Strings.

budaimartin commented on src/hu/elte/txtuml/stdlib/util/HashMap/Hashmap.java:
```diff
@@ -0,0 +1,90 @@
+package hu.elte.txtuml.stdlib.util.HashMap;
```
Package name should be only `hu.elte.txtuml.stdlib.util`.

budaimartin commented on src/hu/elte/txtuml/stdlib/util/HashMap/Hashmap.java:
```diff
@@ -0,0 +1,90 @@
+ * Thus, it's very important not to set the initial capacity too high (or the load factor too low)
+ * if iteration performance is important. 
+ */
+public interface Hashmap extends ExternalClass{
```
Rename this interface `HashMap` so it matches the Java classname.

budaimartin commented on src/hu/elte/txtuml/stdlib/util/HashMap/HashmapImplementation.java:
```diff
@@ -0,0 +1,92 @@
+package hu.elte.txtuml.stdlib.util.HashMap;
```
Correct the package name here as well.

budaimartin commented on src/hu/elte/txtuml/stdlib/util/HashMap/HashmapImplementation.java:
```diff
@@ -0,0 +1,92 @@
+import java.util.Set;
+import hu.elte.txtuml.api.model.Collection;
+
+public class HashmapImplementation implements Hashmap{
```
Rename this class `HashMapImplementation` so it matches the name of the interface.

budaimartin reviewed (changes requested):
It is going to be fine. Please correct package and classnames and delete unnecessary blank lines from `HashMapImplementation` (e.g. line 86).

TatianaCarrascoOrtiz committed:
Add files via upload

Fixing all your requirements
Thank you

TatianaCarrascoOrtiz committed:
Delete Entry.java

TatianaCarrascoOrtiz committed:
Delete EntryImplementation.java

TatianaCarrascoOrtiz committed:
Delete Hashmap.java

TatianaCarrascoOrtiz committed:
Delete HashmapImplementation.java
